1003-button-design-guidance.md

```diff
@@ -0,0 +1,66 @@
+---
+category: component design guidance
+state: draft
+created: 2024-10-15
+updated: 2024-10-22
+---
+
+# Button Design Guidance
+
+The [button](https://clarity.design/documentation/button) component allows users to trigger actions or events with a single click.
+
+## Guidance
+
+Clarity supports solid, outline and flat buttons. These types are provided to give visual distinction between the priority or hierarchy of the buttons in the application.
+
+### Button Types
+
+- **Must** use solid buttons for the primary action that the application is suggesting that the user take.
+- **Must** use outline buttons to indicate a secondary action that complements a primary action.
+- **Should** use outline buttons to reduce visual noise when there are many actions of equal importance on the page.
+- **Should** use flat buttons when a user is expected to take an action within the page.
+- **Should** use a link (or anchor element) in place of a button to navigate the user to a different page.
+- **Should** use a flat button for some tertiary action in the page.
+
+### Button Placement
+
+- **Should** be positioned to help increase their visibility. Clarity follows [Z-pattern and F-pattern](https://clarity.design/documentation/button#placement) for button placement. Use Z-pattern within constrained containers such as modals, and use F-pattern within unconstrained content such as a form on the page itself.
+
+### Button Typography
+
+- The text inside of buttons **must** always be uppercase, to distinguish buttons from the rest of the page content.
+- **Should** not use more than three words inside a button to prevent the button text from wrapping to the next line.
+- **Should** use descriptive language on buttons to help the user understand its intent.
+
+### Button Size
+
+Clarity offers two button sizes : default and compact
+
+- **Should** use compact buttons where smaller buttons are needed to de-emphasize calls to action. Specially when multiple actions of equal importance are available.
+
+### Button with Icons
+
+- **Must not** use two icons in the same button.
```

Comment:
```suggestion
- Design teams **must not** use two icons in the same button, as this can create confusion about the button's purpose and make it harder for users to understand the intended action.
```